src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -6125,6 +6125,93 @@ void QueryAnalyzer::resolveTableFunction(QueryTreeNodePtr & table_function_node,
                 table_function_name);
     }
 
+    QueryTreeNodes result_table_function_arguments;
```

Comment:
I just moved processing `use_structure_from_insertion_table_in_table_functions` after processing table function agruments and calling `table_function_ptr->parseArguments`, because we can use table function arguments (for example, format name).